python/tank/bootstrap/resolver.py

```diff
@@ -230,14 +233,21 @@ def _create_configuration_from_descriptor(self, cfg_descriptor, sg_connection, p
 
     def _get_pipeline_configurations_for_project(self, pipeline_config_name, current_login, sg_connection):
         """
-        Retrieves pipeline configurations from Shotgun that are compatible with the project.
+        Retrieves pipeline configurations from Shotgun that are compatible with the given project.
 
         :param str pipeline_config_name: Name of the pipeline configuration requested for. If ``None``,
             all pipeline configurations from the project will be matched.
         :param str current_login: Only retains non-primary configs from the specified user.
         :param ``shotgun_api3.Shotgun`` sg_connection: Connection to the Shotgun site.
 
-        :returns: A list of pipeline configuration entity dictionaries.
+        :returns: A list of pipeline configuration entity dictionaries with an
+                  extra key 'config_descriptor' added representing a resolved descriptor
+                  instance pointng at the associated configuration to be used for this
+                  pipeline configuration. Please note that this method returns all
+                  pipeline configs matching the given project, meaning that sometimes
+                  the 'config_descriptor' key may be None, for example in the case where
+                  a pipeline configuration is defined for the project for windows and you
+                  are currently running on a mac.
```

Comment:
Nitpick: You are comparing an OS with a computer architecture. `on macOS` is the correct branding for the OS from Apple :p